proposals/csharp-7.0/tuples.md

```diff
@@ -0,0 +1,437 @@
+# Tuples 
+
+This proposal specifies the changes required to the [C# 6.0 (draft) Language specification](../../spec/introduction.md) to support *Tuples* as a new [value type](../../spec/types.md#value-types).
+
+## Changes to [Lexical structure](../../spec/lexical-structure.md)
+
+> The grammar for [Literals](../../spec/lexical-structure.md#literals) is extended to include `tuple_literal`:
+
+```antlr
+literal
+    : boolean_literal
+    | integer_literal
+    | real_literal
+    | character_literal
+    | string_literal
+    | null_literal
+    | tuple_literal // new
+    ;
+
+tuple_literal
+    : '(' tuple_literal_element_list ')'
+    ;
+
+tuple_literal_element_list
+    : tuple_literal_element ',' tuple_literal_element
+    | tuple_literal_element_list ',' tuple_literal_element
+    ;
+
+tuple_literal_element
+    : ( identifier ':' )? expression
+    ;
+```
+
+Note that because it is not a constant expression, a tuple literal cannot be used as default value for an optional parameter.
+
+## Changes to [Types](../../spec/types.md)
+
+> The first paragraph of [Value types](../../spec/types.md#value-types) is replaced with the following text:
+
+A value type is either a `struct` type, an enumeration type, or a tuple type. C# provides a set of predefined struct types called the ***simple types***. The simple types are identified through reserved words.
+
+> The `value_type` grammar is updated to include `tuple_type`:
+
+```antlr
+value_type
+    : struct_type
+    | enum_type
+    | tuple_type // new
+    ;
+
+tuple_type
+    : '(' tuple_type_element_list ')'
+    ;
+    
+tuple_type_element_list
+    : tuple_type_element ',' tuple_type_element
+    | tuple_type_element_list ',' tuple_type_element
+    ;
+    
+tuple_type_element
+    : type identifier?
+    ;
+```
+
+## Additions to [Types](../../spec/types.md)
+
+> The following sections should be added after [Nullable types](../../spec/types.md#nullable-types) (at the end of the current *Value types* section.)
+
+### Tuple types
+
+Tuple types are declared with the following syntax:
```

Comment:
>declared [](start = 16, length = 8)

Tuple types are not "declared".  They are used.  Moreover, your example is not a declaration of a type.  It is an *example* of a method declaration that uses a particular tuple type.

If you are intending to write spec text, you need to clearly distinguish normative (spec) language from examples.  Text such as this is an example, not a normative specification.

Reply:
The first comment is address in #3150 (targeting the feature branch).